Tests/Card Tests/StoredCardComponentTests.swift

```diff
@@ -170,29 +170,30 @@ class StoredCardComponentTests: XCTestCase {
 
         textField.insertText("a")
         textField?.sendActions(for: .editingChanged)
-        XCTAssertEqual(textField.delegate!.textField!(textField, shouldChangeCharactersIn: NSRange(location: 0, length: 1), replacementString: "a"), false)
+        XCTAssertEqual(textField.text, "")
 
-        textField.text = "1"
+        textField.insertText("1")
         textField?.sendActions(for: .editingChanged)
-        XCTAssertEqual(textField.delegate!.textField!(textField, shouldChangeCharactersIn: NSRange(location: 0, length: 1), replacementString: "1"), true)
+        XCTAssertEqual(textField.text, "1")
 
-        textField.text = "11"
+        textField.insertText("1")
         textField?.sendActions(for: .editingChanged)
-        XCTAssertEqual(textField.delegate!.textField!(textField, shouldChangeCharactersIn: NSRange(location: 1, length: 1), replacementString: "1"), true)
+        XCTAssertEqual(textField.text, "11")
 
-        textField.text = "111"
+        textField.insertText("1")
         textField?.sendActions(for: .editingChanged)
-        XCTAssertEqual(textField.delegate!.textField!(textField, shouldChangeCharactersIn: NSRange(location: 2, length: 1), replacementString: "1"), true)
+        XCTAssertEqual(textField.text, "111")
+        
         XCTAssertEqual(payAction.isEnabled, false)
 
-        textField.text = "1111"
+        textField.insertText("1")
         textField?.sendActions(for: .editingChanged)
-        XCTAssertEqual(textField.delegate!.textField!(textField, shouldChangeCharactersIn: NSRange(location: 3, length: 1), replacementString: "1"), true)
+        XCTAssertEqual(textField.text, "1111")
         XCTAssertEqual(payAction.isEnabled, true)
 
-        textField.text = "11111"
+        textField.insertText("1")
         textField?.sendActions(for: .editingChanged)
-        XCTAssertEqual(textField.delegate!.textField!(textField, shouldChangeCharactersIn: NSRange(location: 4, length: 1), replacementString: "1"), false)
+        XCTAssertEqual(textField.text, "1111")
```

Comment:
should be "11111" ?

Reply:
it should be "1111" as it gets prefixed by the validation/formatting logic to a max of 4 characters (for AMEX) and 3 for all other card types